### DISCUSSION TITLE
v2.3.1

### DISCUSSION THREAD

**Discussion by kdrag0n** [kdrag0n/safetynet-fix#195] [Announcements]
### Highlights

- Fixed fingerprint on OxygenOS/ColorOS 12 (@osm0sis)
- Support for Magisk 24+ module updates (@benjibobs)
- Restored support for Android 7

### Other changes

- Spoofed OnePlus OEM unlock status for futureproofing (@osm0sis)
- Minor code improvements

**This version only supports Zygisk (Magisk 24 and newer).**

## Donate

If you found this helpful, please consider supporting development with a **[recurring donation](https://patreon.com/kdrag0n)** for rewards such as early access to updates, exclusive behind-the-scenes development news, and priority support. Alternatively, you can also [buy me a coffee](https://paypal.me/kdrag0ndonate). All support is appreciated :heart: 

<hr /><em>This discussion was created from the release <a href='https://github.com/kdrag0n/safetynet-fix/releases/tag/v2.3.1'>v2.3.1</a>.</em>

**Comment by crea1110** [comment]
Oneplus 7t color12.1 public beta, the fingerprint still can't be used, the play store still prompts that the device has not passed the certification

一加7t  color12.1公测，指纹还是用不了，play商店还是提示设备未通过认证

**Comment by carlito88** [comment]
Fingerprint still doesn't work. 
Device: OnePlus Nord oxygen 12.1 AC2003_11_F.12

**Comment by eulergb7** [comment]
Hellow! I tested on Magisk 23.1 and it works for Google Wallet. I didnt tested on other apps.

**Reply by Bughunter-Geek** [reply]
You're not on Android 13 yet, are you? Because for me it doesn't work on A13 Beta 4.

**Reply by eulergb7** [reply]
Sorry, bro. Actly didnt work for me too. I was wrong. But i have to consider sonething: once i reseted the phone and used an older version of gpay, it worked. Now i dont know if gonna work again.

**Comment by crea1110** [comment]
yes, i still use Android12.1.but i have swiched to Xiaomi,The module works normally and has passed the safety certification of the play store.2022年7月25日 23:10，HennyBro ***@***.***>写道：
You're not on Android 13 yet, are you? Because for me it doesn't work on A13 Beta 4.

**Comment by crea1110** [comment]
Before I used color os based on Android 12.1, it was a public beta version, maybe it was an official problem.  Now I use MIUI12 based on Android 10, the module works fine2022年7月26日 12:23，eulergb7 ***@***.***>写道：
Sorry, bro. Actly didnt work for me too. I was wrong. But i have to consider sonething: once i reseted the phone and used an older version of gpay, it worked. Now i dont know if gonna work again.

**Comment by CaptainDoubleYou** [comment]
Pixel 4 XL here on Android 12 and Magisk 25.2. Gpay isn't working.

**Comment by ViktorShakhrai** [comment]
Xiaomi mi 8 doesn't work,after update to Google wallet 😥

**Reply by CaptainDoubleYou** [reply]
The only way I could get this to work on my Pixel 4 XL was to use Magisk Hide Props Conf module in Magisk along with this one. Give it a shot and see if that does the trick.

**Comment by Displax** [comment]
@CaptainDoubleYou @ViktorShakhrai 
Use my mod. Magisk Hide Props Conf not needed.
https://github.com/Displax/safetynet-fix/releases/tag/v2.3.1-MOD

**Reply by CaptainDoubleYou** [reply]
I tried to use it when I visited the XDA page but it didn't work on my Pixel 4 XL phone.

**Reply by Displax** [reply]
> I tried to use it when I visited the XDA page but it didn't work on my Pixel 4 XL phone.

Then you did something wrong

**Reply by CaptainDoubleYou** [reply]
I'm not sure by what you mean by I did something wrong. You just load the module and disable all other Magisk modules and hope it works. If you can help me troubleshoot the problem then sure by all means. I can help report my findings here.

**Comment by crea1110** [comment]
One plus 7T, color os android 12 of the device I had problems with before. Now I'm using Mi 10Ultra, miui12 Android 10, the 2.3 module works fine on Mi, I don't use 7t anymore, so I can't reproduce the problem，sorry.

从 Windows 版邮件<https://go.microsoft.com/fwlink/?LinkId=550986>发送

发件人: ***@***.***>
发送时间: 2022年8月15日 6:43
收件人: ***@***.***>
抄送: ***@***.***>; ***@***.***>
主题: Re: [kdrag0n/safetynet-fix] v2.3.1 (Discussion #195)


I'm not sure by what you mean by I did something wrong. You just load the module and disable all other Magisk modules and hope it works. If you can help me troubleshoot the problem then sure by all means. I can help report my findings here.

―

**Comment by crea1110** [comment]
I don't think it's your module problem, it's OnePlus' problem

从 Windows 版邮件<https://go.microsoft.com/fwlink/?LinkId=550986>发送

发件人: ***@***.***>
发送时间: 2022年8月15日 6:43
收件人: ***@***.***>
抄送: ***@***.***>; ***@***.***>
主题: Re: [kdrag0n/safetynet-fix] v2.3.1 (Discussion #195)


I'm not sure by what you mean by I did something wrong. You just load the module and disable all other Magisk modules and hope it works. If you can help me troubleshoot the problem then sure by all means. I can help report my findings here.

―

**Comment by tony8077616** [comment]
Is there any possible that update for Riru Version?

**Reply by Displax** [reply]
https://github.com/HuskyDG/riru-integrityfix

**Comment by crea1110** [comment]
I don't know, I haven't used riru for a long time, now I use zygisk

从 Windows 版邮件<https://go.microsoft.com/fwlink/?LinkId=550986>发送

发件人: ***@***.***>
发送时间: 2022年8月16日 20:30
收件人: ***@***.***>
抄送: ***@***.***>; ***@***.***>
主题: Re: [kdrag0n/safetynet-fix] v2.3.1 (Discussion #195)


Is there any possible that update for Riru Version?

―

**Comment by Lyrebird-BBS** [comment]
in my oneplus 8 pro, it works for safety net test, but fails my fingerprint. mine is on Android 12.

**Reply by Displax** [reply]
https://github.com/Displax/safetynet-fix/releases/tag/v2.3.1-MOD

**Reply by Lyrebird-BBS** [reply]
Thanks. However, the mod version does not solve the fingerprint issue.

**Comment by Kuchta2100** [comment]
Occasionally the SafetyNet test fails.  MIUI ROM Xiaomi.eu

**Reply by Kaspe4ycv** [reply]
the same problem
![Screenshot_2022-12-26-09-36-02-872_rikka safetynetchecker](https://user-images.githubusercontent.com/108571994/209492481-b103e9b6-26e0-4050-aad1-7837158e920a.jpg)
![Screenshot_2022-12-26-09-36-15-997_l wi dsnfrsc h e q](https://user-images.githubusercontent.com/108571994/209492485-31632ace-f09b-48bc-8963-d24f8cf8b0b9.jpg)
![Screenshot_2022-12-26-09-37-23-116_l wi dsnfrsc h e q](https://user-images.githubusercontent.com/108571994/209492486-36721715-862d-4f95-895e-c774a7714bb3.jpg)

**Reply by Displax** [reply]
> the same problem
> 

Set SELinux to enforcing

**Reply by Kaspe4ycv** [reply]
> > the same problem
> 
> Set SELinux to enforcing

Can you explain how? Or give a link?

**Reply by Displax** [reply]
> > > the same problem
> > 
> > Set SELinux to enforcing
> 
> Can you explain how? Or give a link?

In the terminal:
su
setenforce 1

**Comment by WUIX69** [comment]
May i ask a question..

2.3.1 seems workin good for me so far, all safetynet checker app that i tried.. shows PASSED at all..

My question is, after Flashing it in Magisk and enabling Zygisk in settings.. should i still need to clear data of GServices and PStore, even its say's Devices certified and Passed all checkers ?

Sorry for nub question 🙃

**Comment by crea1110** [comment]
It seems to be a OnePlus problem, I upgraded my Mi 10ultra to Android 11 and the module still works fine

从 Windows 版邮件<https://go.microsoft.com/fwlink/?LinkId=550986>发送

发件人: ***@***.***>
发送时间: 2022年8月23日 20:56
收件人: ***@***.***>
抄送: ***@***.***>; ***@***.***>
主题: Re: [kdrag0n/safetynet-fix] v2.3.1 (Discussion #195)


in my oneplus 8 pro, it works for safety net test, but fails my fingerprint. mine is on Android 12.

―

**Comment by owocado** [comment]
This really saved me. Thank you alot and have a nice day! ❤️

**Comment by Odin6a** [comment]
Hello all. This is/was working for me. I currently have not tested wallet. I have loaded cards on wallet but have yet to make a purchase. I have also been able to install and run Netflix on my rooted pixel 6a. My concern is that now if I randomly run a safetynet check I fail but Netflix continues to function and wallet continues to hold cards . If I reboot without changes I pass safetynet check.

**Comment by Odin6a** [comment]
Check your settings. I'm using a pixel 6a and my battery is running strong.

**Comment by NaaRK0Z** [comment]
GooglePay not working 😞
